src/replication/ng/scheduler/schedule.go

```diff
@@ -0,0 +1,117 @@
+package schedule
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/goharbor/harbor/src/common/job"
+	common_job "github.com/goharbor/harbor/src/common/job"
+	"github.com/goharbor/harbor/src/common/job/models"
+	"github.com/goharbor/harbor/src/common/utils/log"
+	"github.com/goharbor/harbor/src/core/config"
+	"github.com/goharbor/harbor/src/jobservice/opm"
+	"github.com/goharbor/harbor/src/replication/ng/model"
+)
+
+// DefaultReplicator provides a default implement for Replicator
+type DefaultReplicator struct {
+	client job.Client
+}
+
+// NewDefaultReplicator returns an instance of DefaultReplicator
+func NewDefaultReplicator(client job.Client) *DefaultReplicator {
+	return &DefaultReplicator{
+		client: client,
+	}
+}
+
+// Scheduler to schedule the tasks to transfer resource data
+type Scheduler interface {
+	// Schedule tasks
+	Schedule(srcResources []*model.Resource, destResources []*model.Resource) ([]*model.Task, error)
+	StopExecution(executionID string) error
+}
+
+// Schedule the tasks base on resources
+func (d *DefaultReplicator) Schedule(srcResources []*model.Resource, destResources []*model.Resource) ([]*model.Task, error) {
+	if len(srcResources) != len(destResources) {
+		err := errors.New("srcResources has different length with destResources")
+		log.Errorf(err.Error())
+		return nil, err
+	}
+	var tasks []*model.Task
+	for index, srcResource := range srcResources {
+		destResource := destResources[index]
+		task := &model.Task{}
+		task.ResourceType = srcResource.Type
+		task.StartTime = time.Now().UTC()
+		src, err := json.Marshal(srcResource)
+		if err != nil {
+			log.Errorf("failed to marshal the srcResource of %v.err:%s!", srcResource, err.Error())
+			task.Status = "Error"
+			tasks = append(tasks, task)
+			continue
+		}
+		task.SrcResource = string(src)
+		dest, err := json.Marshal(destResource)
+		if err != nil {
+			log.Errorf("failed to marshal the destResource of %v.err:%s!", destResource, err.Error())
+			task.Status = "Error"
+			tasks = append(tasks, task)
+			continue
+		}
+		task.DstResource = string(dest)
+		task.Status = "Initial"
```

Comment:
we don't design a Initial status for task, maybe should use Pending?
i feel a little confused about the translate of the status.
init(Pending)->js accept and return a jobID(running)->run success(Success)
                                                                               ->run failed(Failed)
                                                                               ->retrying(Retrying) this not support now
                                                                               ->stoped(Stopped)
@steven-zou @ywk253100

Reply:
remove the task status from scheduler, leave the status management to flow controller